packages/client-react-streaming/src/transportedQueryRef.ts

```diff
@@ -164,43 +194,69 @@ function createTransportedQueryRef<
-    ];
-    hydrationCache.set(queryRef, { cacheKey });
-    const internalQueryRef = getSuspenseCache(client).getQueryRef(
-      cacheKey,
-      () =>
-        client.watchQuery({
-          ...hydratedOptions,
-          fetchPolicy: "network-only",
-          context: skipDataTransport(
-            readFromReadableStream(stream.pipeThrough(new JSONDecodeStream()), {
-              ...hydratedOptions.context,
-              queryDeduplication: true,
-            })
-          ),
-        })
+  const hydratedOptions = deserializeOptions(options);
+  const internalQueryRef = getInternalQueryRef(client, hydratedOptions, {
+    ...hydratedOptions,
+    fetchPolicy: "network-only",
+    context: skipDataTransport(
+      readFromReadableStream(stream.pipeThrough(new JSONDecodeStream()), {
+        ...hydratedOptions.context,
+        queryDeduplication: true,
+      })
+    ),
+  });
+  Object.assign(queryRef, wrapQueryRef(internalQueryRef));
+}
+
+function getInternalQueryRef(
+  client: ApolloClient,
+  userOptions: ApolloClient.WatchQueryOptions,
+  initialFetchOptions: ApolloClient.WatchQueryOptions
+) {
+  if (__DEV__) {
+    // this would be a bug in our code, `initialFetchOptions` should always be derived
+    // from `userOptions` and never change `nextFetchPolicy` or we would need additional
+    // logic to handle that case
+    invariant(
+      userOptions.nextFetchPolicy === initialFetchOptions.nextFetchPolicy
     );
-    Object.assign(queryRef, wrapQueryRef(internalQueryRef));
   }
+  // create with `userOptions` so internals like `initialFetchPolicy` are set correctly
+  const observable = client.watchQuery(userOptions);
+  // this might have filled in some defaults, so we need to capture them
+  const optionsAfterCreation = {
+    // context might still be `undefined`, so we need to make sure the property is at least present
+    // `undefined` won't merge in as `applyOptions` uses `compact`, so we use an empty object instead
+    context: {},
+    ...observable.options,
+  };
+  // apply `initialFetchOptions` for the first fetch
+  observable.applyOptions(initialFetchOptions);
+  // `new InternalQueryReference` calls `observable.subscribe` immediately, so the `initialFetchOptions` are applied to that call
+  const internalQueryRef = new InternalQueryReference(observable, {
+    autoDisposeTimeoutMs:
+      client.defaultOptions.react?.suspense?.autoDisposeTimeoutMs,
+  });
+  // set the options for any future `reobserve` calls back to `userOptions` directly after that
+  observable.applyOptions({
+    ...optionsAfterCreation,
+    fetchPolicy:
+      observable.options.fetchPolicy !== initialFetchOptions.fetchPolicy
+        ? // if `fetchPolicy` was changed from `initialFetchOptions`, `nextFetchPolicy` has been applied and we're not going to touch it
+          observable.options.fetchPolicy
+        : // otherwise we restore the `userOptions.fetchPolicy` for future fetches
+          optionsAfterCreation.fetchPolicy,
```

Comment:
```suggestion
      observable.options.fetchPolicy === initialFetchOptions.fetchPolicy
        ? // restore `userOptions.fetchPolicy` for future fetches
          optionsAfterCreation.fetchPolicy,
        : // otherwise `fetchPolicy` was changed from `initialFetchOptions`, `nextFetchPolicy` has been applied and we're not going to touch it
          observable.options.fetchPolicy,
```

Similarly, I find it easier to use the positive connotation since this makes it a bit more obvious that you're after restoring user provided options with this block of code.